File: pocket-android-aion/src/main/java/network/pokt/aion/util/HexStringUtil.java
package network.pokt.aion.util;

public class HexStringUtil {

    private static final String ZERO_X = "0x";

    public static String prependZeroX(String hex) {
        if (hasZeroHexPrefix(hex)) {
            return hex;
        } else {
            return ZERO_X + hex;
        }
    }

    public static String removeLeadingZeroX(String hex) {
        if (hasZeroHexPrefix(hex)) {
            if (hex.length() > ZERO_X.length()) {
                return hex.substring(ZERO_X.length(), hex.length());
            } else {
                return "";
            }
        } else {
            return hex;
        }
    }

    public static boolean hasZeroHexPrefix(String hex) {
        return hex.length() >= ZERO_X.length() && hex.substring(0, 2).equalsIgnoreCase(ZERO_X);
    }

}

File: pocket-android-aion/src/main/java/network/pokt/pocketaion/util/HexStringUtil.kt
package network.pokt.pocketaion.util

class HexStringUtil {

    private val ZERO_X = "0x"

    fun prependZeroX(hex: String): String {
        return if (hasZeroHexPrefix(hex)) {
            hex
        } else {
            ZERO_X + hex
        }
    }

    fun removeLeadingZeroX(hex: String): String {
        return if (hasZeroHexPrefix(hex)) {
            if (hex.length > ZERO_X.length) {
                hex.substring(ZERO_X.length, hex.length)
            } else {
                ""
            }
        } else {
            hex
        }
    }

    fun hasZeroHexPrefix(hex: String): Boolean {
        return if (hex.length >= ZERO_X.length) hex.substring(0, 2).equals(ZERO_X, ignoreCase = true) else false
    }
}